File: Back-end/TransactionService/Models/Transaction.cs
namespace TransactionService.Models
{
    public enum TransactionType
    {
        Purchase = 1,
        Sale = 2
    }

    public class Transaction
    {
        public int Id { get; set; }
        public DateTime DateUtc { get; set; }
        public TransactionType Type { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalPrice { get; set; }
        public string? Detail { get; set; }
    }
}

File: Front-end/micro-services/src/app/shared/models/transaction.ts
export type TransactionType = 1 | 2; // 1=Purchase, 2=Sale

export interface Transaction {
  id: number;
  dateUtc: string;
  type: TransactionType;
  productId: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  detail: string | null;
}
